File: Structural-Patterns/Decorator/Java/Implementations/SwordDecorator.java
package Implementations;

import java.util.List;

import Interfaces.CharacterDecorator;
import Interfaces.ICharacter;

public class SwordDecorator extends CharacterDecorator{

    public SwordDecorator(ICharacter character) {
        super(character);
    }

    @Override
    public int getAttackPower(){
        return super.getAttackPower() + 10;
    }

    @Override
    public String getDescription(){
        return super.getDescription() + " , weilding a mighty sword";
    }

    @Override
    public List<String> getAbilities(){
        List<String> abilities = super.getAbilities();
        abilities.add("Sword Hit");
        return abilities;
    }
}

File: Structural-Patterns/Decorator/C#/Decorator/Implementations/SwordDecorator.cs
public class SwordDecorator : CharacterDecorator{

    public SwordDecorator(ICharacter character) : base(character) {}

    public override int GetAttackPower(){
        return base.GetAttackPower() + 10;
    }

    public override String GetDescription(){
        return base.GetDescription() + " , weilding a mighty sword";
    }

    public override List<String> GetAbilities(){
        List<String> abilities = base.GetAbilities();
        abilities.Add("Sword Hit");
        return abilities;
    }
}
